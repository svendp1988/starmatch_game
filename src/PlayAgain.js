import React, { Component } from "react";

class PlayAgain extends Component {
    render() {
        return (
            <div className="game-done">
                <div
                    className="message"
                    style={{ color: this.props.gameStatus === "lost" ? "red" : "green"}}
                >
                    {this.props.gameStatus === "lost" ? "Game Over": "Well played!"}
                </div>
                <button onClick={this.props.onClick}>Play Again</button>
            </div>
        )
    }
}

export default PlayAgain;
