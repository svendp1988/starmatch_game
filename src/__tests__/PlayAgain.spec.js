import React from 'react';
import {shallow, mount} from 'enzyme';
import PlayAgain from "../PlayAgain";
import renderer from 'react-test-renderer';




describe("PlayAgain", () => {
    it('should match snapshot', function () {
        const props = {
            gameStatus: "lost"
        }
        const tree = renderer.create(
            <PlayAgain {...props}/>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have a red colored message "Game Over" when the game is lost', function () {
        const props = {
            gameStatus: "lost"
        }
        const wrapper = shallow(<PlayAgain {...props}/>);
        expect(wrapper.find(".message").text()).toEqual("Game Over");
        expect(wrapper.find(".message").props().style).toEqual({ color: 'red' });
    });

    it('should have a green colored message "Well played!" when the game is won', function () {
        const props = {
            gameStatus: "won"
        }
        const wrapper = shallow(<PlayAgain {...props}/>);
        let messageBox = wrapper.find(".message");
        expect(messageBox.text()).toEqual("Well played!");
        expect(messageBox.props().style).toEqual({ color: "green" });
    });
})
