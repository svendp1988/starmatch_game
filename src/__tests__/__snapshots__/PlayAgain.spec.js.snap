// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayAgain should match snapshot 1`] = `
<div
  className="game-done"
>
  <div
    className="message"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Game Over
  </div>
  <button>
    Play Again
  </button>
</div>
`;
